### DISCUSSION TITLE
Cron Job not running in production

### DISCUSSION THREAD

**Discussion by christian-valadez** [vercel/community#2000] [Help]
### Summary

My cron job is not being invoked in production. If I go to `localhost:3000/api/cron/startGiveaway`, the server responds correctly. But Vercel is not running it every hour when deployed to production.

```json
{
  "crons": [
    {
      "path": "/api/cron/startGiveaway",
      "schedule": "0 * * * *"
    }
  ],
}

```

My file path is: `apps/nextjs/pages/api/cron/startGiveaway.api.ts. 

My `next.config.mjs` uses pageExtensions: 

```
 pageExtensions: ['page.tsx', 'page.ts', 'page.jsx', 'page.js', 'api.js', 'api.ts', 'api.tsx'],
```

What's going on?

**Comment by amyegan** [comment]
Hi, Christian! I have a few questions to help us figure out what is happening:
- If you visit `/api/cron/startGiveaway` on your production or preview deployment URLs, does the function work as expected?
- Does the project belong to a Hobby account or a Pro team?
- Do you see any errors in the [cron jobs logs](https://vercel.com/docs/cron-jobs#cron-jobs-list) or [runtime logs](https://vercel.com/docs/concepts/observability/runtime-logs#)?

**Comment by fightingmonk** [comment]
@amyegan I'm also seeing issues with a daily cron job not being invoked. I set up the cron job a few days ago and it doesn't seem to have ever run.

Function works as expected when manually invoked, Hobby account, no errors in cron job or runtime logs.

vercel.json
```json
{
    "crons": [
        {
            "path": "/api/integration/daily-marketing",
            "schedule": "0 16 * * *"
        }
    ]
}
```

I see the job correctly listed in the project's  `Cron Jobs` ui but there's no evidence in Logs that the function is being invoked automatically.

When I visit `/api/integration/daily-marketing` in a browser or with curl the api function executes as expected, and when I click the `Run` button in the `Cron Jobs` view it also executes and completes successfully.

**Reply by amyegan** [reply]
Hi @fightingmonk. Aside from the logs, are you not seeing the effect you expected from the serverless function running? Don't forget that cron jobs are scheduled for UTC time, which I know has thrown me off in the past. And remember that Hobby accounts have a limit of 2 cron jobs, each triggered once a day, with [imprecise scheduling](https://vercel.com/docs/cron-jobs#hobby-scheduling-limits).

If you're able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), an example deployment link, or the ID of the deployment that should currently run the cron job, then I can take a closer look

**Reply by fightingmonk** [reply]
@amyegan thanks for the quick followup!

> are you not seeing the effect you expected from the serverless function running? 

Correct - the function generates logs and sends email through sendgrid, and none of the expected behaviors are happening. I added the cron config for a single once-a-day job on April 10, so it should have run yesterday and today at 1700 UTC.

> If you're able to share the ID of the deployment that should currently run the cron job, then I can take a closer look

The current production deployment id is `9UtiuKWhMLttbbQNBT2aht7k3ERe`. Thanks in advance for looking into this
